### PR TITLE
feat: data set list view

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [DHIS2-18283](https://dhis2.atlassian.net/browse/DHIS2-18283) and [DHIS2-18284](https://dhis2.atlassian.net/browse/DHIS2-18284)
---

### Description
This PR displays the dataset list view with proper filteration.

---

### Checklist

-   [ ] Dataset list View

---

[DHIS2-18283]: https://dhis2.atlassian.net/browse/DHIS2-18283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-18284]: https://dhis2.atlassian.net/browse/DHIS2-18284?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/sectionList/filters/DynamicFilters.tsx [hunk 1]
```diff
@@ -7,24 +7,24 @@ import {
...
 type FilterKeyToComponentMap = Partial<Record<ConfigurableFilterKey, React.FC>>
 
 const filterKeyToComponentMap: FilterKeyToComponentMap = {
     category: Categoryfilter,
+    formType: FormTypeFilter,
     indicatorType: IndicatorFilter,
     categoryOption: CategoryOptionFilter,
     categoryCombo: CategoryComboFilter,
     categoryOptionGroup: CategoryOptionGroupFilter,
-    dataSet: DataSetFilter,
```
Comment (Birkbjo) [inline src/components/sectionList/filters/DynamicFilters.tsx:27]: Why is this removed? This is used in the `dataElement`-list.

**Review by Birkbjo — CHANGES_REQUESTED** [review]
Looks good, but lets not remove `dataSet`-filter? This is used to filter `dataElements` by `dataSet` in the `dataElement`-list.

File: src/components/sectionList/filters/DynamicFilters.tsx (hunk 1 above)
Comment (Chisomchima) [reply src/components/sectionList/filters/DynamicFilters.tsx:27]: Oh, I actually thought I was the one who added it. will add it back.

**Review by Birkbjo — APPROVED** [review]
